src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -9,6 +10,42 @@ public class RegistrationServiceImpl implements RegistrationService {
 
     @Override
     public User register(User user) {
-        return null;
+        if (user == null) {
+            throw new RegistrationException("User can't be null!");
+        }
+        checkLogin(user.getLogin());
+        checkPassword(user.getPassword());
+        checkAge(user.getAge());
+        if (storageDao.get(user.getLogin()) != null) {
+            throw new RegistrationException("User already exists!");
+        }
+        return storageDao.add(user);
+    }
+
+    private void checkPassword(String password) {
+        if (password == null) {
+            throw new RegistrationException("Password can't be null!");
+        }
+        if (password.length() < 6) {
+            throw new RegistrationException("Password must be at least 6 characters long!");
+        }
```

Comment:
Make 6 constants and use it for conditional and exception messages. Make the same for age